src/Mod/Path/PathScripts/PostUtilsParse.py

```diff
@@ -422,44 +404,32 @@ def parse(values, pathobj):
                         ):
                             continue
                         else:
-                            pos = Units.Quantity(
-                                c.Parameters[param], FreeCAD.Units.Length
-                            )
-                            outstring.append(
-                                param
-                                + format(
-                                    float(pos.getValueAs(values["UNIT_FORMAT"])),
-                                    axis_precision_string,
-                                )
+                            pos = Units.Quantity(c.Parameters[param], Units.Length)
+                            param_num = format(
+                                float(pos.getValueAs(values["UNIT_FORMAT"])),
+                                axis_precision_string,
                             )
+                            outstring.append(f"{param}{param_num}")
 
             if values["OUTPUT_ADAPTIVE"]:
                 if adaptiveOp and command in values["RAPID_MOVES"]:
                     if opHorizRapid and opVertRapid:
                         if "Z" not in c.Parameters:
-                            outstring.append(
-                                "F"
-                                + format(
-                                    float(
-                                        opHorizRapid.getValueAs(
-                                            values["UNIT_SPEED_FORMAT"]
-                                        )
-                                    ),
-                                    axis_precision_string,
-                                )
+                            param_num = format(
+                                float(
+                                    opHorizRapid.getValueAs(values["UNIT_SPEED_FORMAT"])
+                                ),
+                                axis_precision_string,
                             )
+                            outstring.append(f"F{param_num}")
                         else:
-                            outstring.append(
-                                "F"
-                                + format(
-                                    float(
-                                        opVertRapid.getValueAs(
-                                            values["UNIT_SPEED_FORMAT"]
-                                        )
-                                    ),
-                                    axis_precision_string,
-                                )
+                            param_num = format(
+                                float(
+                                    opVertRapid.getValueAs(values["UNIT_SPEED_FORMAT"])
+                                ),
+                                axis_precision_string,
                             )
+                            outstring.append(f"F{param_num}")
```

Comment:
```suggestion
            if (
                values["OUTPUT_ADAPTIVE"]
                and adaptiveOp
                and command in values["RAPID_MOVES"]
                and opHorizRapid
                and opVertRapid
            ):
                if "Z" not in c.Parameters:
                    param_num = format(
                        float(opHorizRapid.getValueAs(values["UNIT_SPEED_FORMAT"])),
                        axis_precision_string,
                    )
                else:
                    param_num = format(
                        float(opVertRapid.getValueAs(values["UNIT_SPEED_FORMAT"])),
                        axis_precision_string,
                    )
                outstring.append(f"F{param_num}")
```
Gives bonus points

Reply:
That is much cleaner.  I will make that change.
This is "inherited code" that I am in the process of refactoring, so there are quite a few places where I have left the code the way it was while I concentrate on other areas.   I still have quite a way to go to get it all cleaned up and organized better.
Thanks for spotting that opportunity!